src/linuxforhealth/csvtofhir/pipeline/tasks.py

```diff
@@ -1037,3 +1038,88 @@ def append_list(
         if "tmp" in data_frame.columns:
             data_frame.drop(["tmp"], axis=1, inplace=True)
     return data_frame
+
+
+def join_data(
+    data_frame: DataFrame,
+    secondary_data_source: str,
+    join_type: str,
+    join_on: str,
+    source_type: str = "csv",
+    reader_params: Optional[Dict] = None
+) -> DataFrame:
+    """
+    takes a secondary file (csv or fixed width) and joins the supplimentary data with the primary
+    dataframe based on some common joining key.
+
+    Example:
+    primary dataframe Patient:
+    first name, last name, gender, MRN
+
+    supplimentary data:
+    MRN, residentail address
+
+    after a join, the primary dataframe should contain the residential address of each patient 
+    as long as it was present in the supplimentary data.
+
+    :param data_frame: The input DataFrame
+    :param secondary_data_source: path to the secondary data file. Can be relative to the data-contract dictionary or absolute.
+    Also supports http, ftp, s3 and gs paths
```

Comment:
Confirming that pandas is supporting http, ftp, s3, etc paths and that the new "optimized-streaming" extra is not required.